private-templates-service/client/src/components/AdaptiveCardPanel/AdaptiveCardPanel.tsx

```diff
@@ -48,13 +69,17 @@ class AdaptiveCardPanel extends React.Component<Props> {
           <TemplateStateWrapper style={{ justifyContent: "center" }}>
             <StatusIndicator
               state={
-                template.isLive
-                  ? PostedTemplate.StateEnum.Live
-                  : PostedTemplate.StateEnum.Draft
+                (template.instances && template.instances[0] && template.instances[0].state)
```

Comment:
I feel that the case where `template.instances[0].state === undefined` should be handled in your `retrieveStateValue` function. this could be something like `<StatusIndicator state={retrieveStateValue(template)} ...`